### PR TITLE
[SPARK-133] Restructure test folder

### PR DESCRIPTION
**Pull request by Eearslya**

### CODE REVIEW

File: tests/unit/test_autocorrect.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 from src.packages.utils import correct_system_name
 
-pytestmark = pytest.mark.autocorrect
+pytestmark = [pytest.mark.unit, pytest.mark.autocorrect]
```
Comment (theunkn0wn1) [inline tests/unit/test_autocorrect.py:16]: [question] can i get a documentation link for this syntax?

**Review by theunkn0wn1 — APPROVED** [review]
LGTM

File: tests/unit/test_autocorrect.py (hunk 1 above)
Comment (Eearslya) [reply tests/unit/test_autocorrect.py:16]: https://docs.pytest.org/en/latest/example/markers.html#marking-whole-classes-or-modules

>or if you need to use multiple markers you can use a list:
>```python
>import pytest
>
>class TestClass(object):
>    pytestmark = [pytest.mark.webtest, pytest.mark.slowtest]
>```

Comment (theunkn0wn1) [reply tests/unit/test_autocorrect.py:16]: awesome, thanks! wasn't aware test marks could be combined like this!

**Pull request merged**